From: Tio Zhang <tiozhang@didiglobal.com>
Subject: fork: remove #ifdef CONFIG_LOCKDEP in copy_process()
Date: Wed, 20 Aug 2025 18:18:46 +0800

lockdep_init_task() is defined as an empty when CONFIG_LOCKDEP is not set.
So the #ifdef here is redundant, remove it.

Link: https://lkml.kernel.org/r/20250820101826.GA2484@didi-ThinkCentre-M930t-N000
Signed-off-by: Tio Zhang <tiozhang@didiglobal.com>
Cc: Kees Cook <kees@kernel.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 kernel/fork.c |    2 --
 1 file changed, 2 deletions(-)

--- a/kernel/fork.c~fork-remove-ifdef-config_lockdep-in-copy_process
+++ a/kernel/fork.c
@@ -2127,9 +2127,7 @@ __latent_entropy struct task_struct *cop
 
 	p->pagefault_disabled = 0;
 
-#ifdef CONFIG_LOCKDEP
 	lockdep_init_task(p);
-#endif
 
 	p->blocked_on = NULL; /* not blocked yet */
 
_
